### PR TITLE
Adding GH Actions workflow for version checking

### PR DESCRIPTION
**Pull request by JSchoenbachler**
This adds a GH Actions workflow that does the following:
1. Uses R to read lab package versions on CRAN and within the drat repo.
2. Calculates which drat packages meet a version difference threshold (>5 patch releases, >2 minor releases, or 1 major release).
3. Closes any open GH Issues labeled `cran-update`
4. Creates an issue for each package exceeding one of the version difference thresholds with the label of `cran-update`.

### CODE REVIEW

File: .github/workflows/cran-check.yaml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+          dratPackages = data.table(filename = list.files(file.path('src', 'contrib')))
+          dratPackages[, filename := gsub('.tar.gz', '', filename, fixed = TRUE)]
+          dratPackages = dratPackages[, c('name', 'version') := tstrsplit(filename, '_', fixed = TRUE)]
+          dratPackages = dratPackages[
+            !is.na(version), .(drat_version = max(version)), by = name]
+
+          # Establish version levels and split out into separate columns
+          versionLevels = c('major', 'minor', 'patch', 'dev')
+          dratPackages[, (paste0('drat_', versionLevels)) :=
+                        tstrsplit(drat_version, '.', fixed = TRUE, type.convert = TRUE)]
```
Comment (jakejh) [inline .github/workflows/cran-check.yaml]: This line should start only two spaces past the previous, similar to lines 34 and 35.

File: .github/workflows/cran-check.yaml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+          # Read in packages availability on CRAN and merge with drat data.table
+          cranPackages = as.data.table(available.packages(repos = 'https://cran.rstudio.com/'))
+          cranPackages = cranPackages[, .(name = Package, cran_version = Version)]
+          packages = merge(dratPackages, cranPackages, by = 'name')
+
+          # Split out CRAN version levels (assuming no dev level exists on CRAN)
+          packages[, (paste0('cran_', versionLevels[1:3])) :=
+                    tstrsplit(cran_version, '.', fixed = TRUE, type.convert = TRUE)]
```
Comment (jakejh) [inline .github/workflows/cran-check.yaml]: This line should start only two spaces past the previous, similar to lines 34 and 35.

File: .github/workflows/cran-check.yaml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+          packages[, needs_cran_update := 0]
+          packages[
+            drat_major > cran_major |
+            drat_minor > cran_minor + 2 |
+            drat_patch > cran_patch + 5,
+            needs_cran_update := 1]
+
+          packages = packages[needs_cran_update == 1]
+
+          needsUpdatesOutput = paste0(packages$name, '---', packages$drat_version, '---', packages$cran_version, collapse = '", "')
```
Comment (jakejh) [inline .github/workflows/cran-check.yaml]: Please add line breaks as so:

```r
needsUpdatesOutput = paste0(
  packages$name, '---', packages$drat_version,
  '---', packages$cran_version, collapse = '", "')
```

File: .github/workflows/cran-check.yaml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+          needsUpdatesOutput = paste0('["', needsUpdatesOutput, '"]')
+
+          s = sprintf('echo \'::set-output name=value::%s\'', needsUpdatesOutput)
```
Comment (jakejh) [inline .github/workflows/cran-check.yaml]: Can we change this to below for readability?

```r
s = sprintf("echo '::set-output name=value::%s'", needsUpdatesOutput)
```

File: .github/workflows/cran-check.yaml [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+  create_issue:
+    needs: [ check_version_number, close_open_cran ]
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      issues: write
+    strategy:
+      matrix:
+        value: ${{fromJson(needs.check_version_number.outputs.matrix)}}
```
Comment (jakejh) [inline .github/workflows/cran-check.yaml]: For consistency, add whitespace around double braces.

File: .github/workflows/cran-check.yaml [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+  close_open_cran:
+    name: Close all issues labeled "cran-update"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Find all cran update issues
+        id: issues
+        uses: lee-dohm/select-matching-issues@v1.2.0
+        with:
+          query: 'label:cran-update is:open'
```
Comment (jakejh) [inline .github/workflows/cran-check.yaml]: For consistency, probably use double quotes.

File: .github/workflows/cran-check.yaml [hunk 7]
```diff
@@ -0,0 +1,125 @@
+# Workflow derived from https://github.com/r-lib/actions/tree/master/examples
```
Comment (jakejh) [inline .github/workflows/cran-check.yaml]: Are these first two lines really still applicable?

**Review by jakejh — CHANGES_REQUESTED** [review]

**Comment by JSchoenbachler** [comment]
@jakejh Changes have been made per your requests.

**Review by jakejh — APPROVED** [review]

**Comment by JSchoenbachler** [comment]
@jakejh I actually don't have write access to this repository, could you merge?

**Pull request merged**